doc/user_guide/dynamo0p3.rst

```diff
@@ -753,19 +748,57 @@ later in this section (see :ref:`dynamo0.3-kernel-valid-access`).
   Kernel means that synchronisation or colouring is required for
   parallel runs.
 
+* ``GH_INC`` indicates that different iterations of a Kernel make
+  contributions to shared values. For example, values at cell faces
+  may receive contributions from cells on either side of the
+  face. This means that such a Kernel needs appropriate
+  synchronisation (or colouring) to run in parallel.
+
+* ``GH_READINC`` indicates that the data is first read and then
+  subsequently incremented. Therefore this is equivalent to a
+  ``GH_READ`` followed by a ``GH_INC``.
+
 * ``GH_SUM`` is an example of a reduction and is the only reduction
   currently supported in PSyclone. This metadata indicates that values
   are summed over calls to Kernel code.
 
 For example::
 
-  type(arg_type) :: meta_args(4) = (/                                  &
-       arg_type(GH_SCALAR, GH_REAL, GH_SUM),                           &
-       arg_type(GH_FIELD, GH_INTEGER, GH_INC, ... ),                   &
-       arg_type(GH_FIELD*3, GH_REAL, GH_WRITE, ... ),                  &
-       arg_type(GH_OPERATOR, GH_REAL, GH_READ, ...)                    &
+  type(arg_type) :: meta_args(6) = (/                            &
+       arg_type(GH_OPERATOR, GH_REAL,    GH_READ,      ... ),    &
+       arg_type(GH_FIELD*3,  GH_REAL,    GH_WRITE,     ... ),    &
+       arg_type(GH_FIELD,    GH_REAL,    GH_READWRITE, ... ),    &
+       arg_type(GH_FIELD,    GH_INTEGER, GH_INC,       ... ),    &
+       arg_type(GH_FIELD,    GH_REAL,    GH_READINC,   ... ),    &
+       arg_type(GH_SCALAR,   GH_REAL,    GH_SUM)                 &
        /)
 
+.. warning:: It is important that ``GH_INC`` is not incorrectly used
+  in place of ``GH_READINC`` as it could result in the reading of data
+  from a dirty outermost halo when run in parallel, giving incorrect
+  results. The reason for this is that PSyclone does not add a halo
+  exchange for the outermost modified halo level of a field before a
+  loop that contains a ``GH_INC`` access to that field i.e. a loop
+  iterating to the level-n halo will result in a halo exchange to the
+  level n-1 halo being added before the loop (which means no halo
+  exchange is added when `n==1`). The reason this can be performed is
+  because any computation performed in the outermost halo will be
+  incorrect (will only compute partial sums) and PSyclone therefore sets
+  this halo level to dirty after the loop has completed. There is therefore
```

Comment:
I would suggest commas around "therefore", i.e. "There is, therefore, no reason..."

Reply:
O,K,